packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -249,6 +241,58 @@ export interface GraphQLApiProps {
 
 }
 
+/**
+ * A IamResource Class for IAM Permissions
+ */
+export class IamResource {
+  /**
+   * Generate the resourceNames given custom arns
+   *
+   * @param arns The custom arns that need to be permissioned
+   */
+  public static custom(...arns: string[]): IamResource {
+    return new IamResource(arns);
+  }
+
+  /**
+   * Generate the resourceNames given a type and fields
+   *
+   * @param type The type that needs to be allowed
+   * @param fields The fields that need to be allowed, if empty then ARN set to `type/*`
```

Comment:
So empty means "grant permissions to ALL fields", right? It might be worth it to say that explicitly in the docs.